### ISSUE TITLE
How to mask latency in `brew install`

### ISSUE THREAD

**Issue by dimaqq** [Homebrew/brew#7030]
### Why?

Whenever I install something, my experience is this:
```sh
> brew install github/gh/gh
Updating Homebrew...

# wait..

# wait ..

# wait..

# a few minutes later

==> Auto-updated Homebrew!
Updated 5 taps (...output...).
==> New Formulae
#  ...tons of output...
==> Deleted Formulae
# ...output...
==> Tapping github/gh
# ...some output...
🍺  /usr/local/Cellar/gh/0.5.4: 9 files, 13.4MB, built in 10 seconds
```

### A detailed description of the proposed feature

I would really like to mask the `Updating Homebrew...` which is essentially latency / productivity tax.
That is when I install something, it's because I want to use it. Or in layman's terms, I want it **now**.

Likewise, while it's awesome to report what formulae were updated, there's too much output.

Is it realistic to update homebrew asynchronously, e.g.?
* start installing the package as we know it
* meanwhile update homebrew
* when both are complete, if we now know that the package has newer version, then reinstall it

Or is it possible to update homebrew periodically, so that it's already "updated" when I `brew install`?

Or could `Updating` part download a hosted/mirrored file that essentially contains "package_name → newest version", so that if what I'm installing (and it's deps) version(s) are what brew already has, it can skip (or postpone) the update phase?

**Comment by MikeMcQuaid** [comment]
> That is when I install something, it's because I want to use it. Or in layman's terms, I want it **now**.

You also probably want the latest, working version which you aren't sure you are going to get without the `brew update`.

> Is it realistic to update homebrew asynchronously

No, unfortunately not.



> Or is it possible to update homebrew periodically, so that it's already "updated" when I `brew install`?

Yes, you could do this in the background yourself if desired.

> Or could `Updating` part download a hosted/mirrored file that essentially contains "package_name → newest version", so that if what I'm installing (and it's deps) version(s) are what brew already has, it can skip (or postpone) the update phase?

No.

---

Please always fill out the issue template. 

`HOMEBREW_AUTO_UPDATE_SECS` may also be of use to you (see `brew man`).
